File: java/nanocontainer/src/test/org/nanocontainer/script/AbstractScriptedContainerBuilderTestCase.java
/*****************************************************************************
 * Copyright (C) NanoContainer Organization. All rights reserved.            *
 * ------------------------------------------------------------------------- *
 * The software in this package is published under the terms of the BSD      *
 * style license a copy of which has been included with this distribution in *
 * the LICENSE.txt file.                                                     *
 *                                                                           *
 * Original code by                                                          *
 *****************************************************************************/
/**
 * @author Aslak Helles&oslash;y
 * @version $Revision$
 */
package org.nanocontainer.script;

import org.jmock.MockObjectTestCase;
import org.picocontainer.PicoContainer;
import org.picocontainer.defaults.ObjectReference;
import org.picocontainer.defaults.SimpleReference;

public abstract class AbstractScriptedContainerBuilderTestCase extends MockObjectTestCase {
    private ObjectReference containerRef = new SimpleReference();
    private ObjectReference parentContainerRef = new SimpleReference();

    protected PicoContainer buildContainer(ScriptedContainerBuilder builder, PicoContainer parentContainer, Object scope) {
        parentContainerRef.set(parentContainer);
        builder.buildContainer(containerRef, parentContainerRef, scope, true);
        return (PicoContainer) containerRef.get();
    }
}

File: trunk/dotnet/nanocontainer/src/Test/Script/AbstractScriptedContainerBuilderTestCase.cs
using System.Collections;
using System.IO;
using System.Reflection;
using NanoContainer.Script;
using PicoContainer.Core;
using PicoContainer.Defaults;

namespace Test.Script
{
	public abstract class AbstractScriptedContainerBuilderTestCase
	{
		protected IPicoContainer BuildContainer(ScriptedContainerBuilder builder, IPicoContainer parentContainer, IList assemblies)
		{
			SimpleReference sr = new SimpleReference();
			SimpleReference pr = new SimpleReference();
			pr.Set(parentContainer);
			builder.BuildContainer(sr, pr, assemblies);
			return (IPicoContainer) sr.Get();
		}

		public StreamReader GetStreamReader(string scriptName)
		{
			Stream strm = Assembly.GetExecutingAssembly().GetManifestResourceStream(scriptName);

			return new StreamReader(strm);
		}
	}
}